src/back/routes/badges.ts

```diff
@@ -1,17 +1,38 @@
+import { WithAuth, auth } from 'decentraland-gatsby/dist/entities/Auth/middleware'
+import RequestError from 'decentraland-gatsby/dist/entities/Route/error'
 import handleAPI from 'decentraland-gatsby/dist/entities/Route/handle'
 import routes from 'decentraland-gatsby/dist/entities/Route/routes'
 import { Request } from 'express'
 
 import { UserBadges } from '../../entities/Badges/types'
+import isDebugAddress from '../../entities/Debug/isDebugAddress'
 import { BadgesService } from '../../services/BadgesService'
 import { validateAddress } from '../utils/validations'
 
 export default routes((router) => {
+  const withAuth = auth()
   router.get('/badges/:address/', handleAPI(getBadges))
+  router.post('/badges/airdrop/', withAuth, handleAPI(airdropBadges))
 })
 
 async function getBadges(req: Request<{ address: string }>): Promise<UserBadges> {
   const address = req.params.address
   validateAddress(address)
   return await BadgesService.getBadges(address)
 }
+
+async function airdropBadges(req: WithAuth): Promise<string> {
+  const user = req.auth!
+  const recipients: string[] = req.body.recipients
+  const badgeSpecCId = req.body.badgeSpecCid
+
+  if (!user || !isDebugAddress(user)) {
```

Comment:
The `!user` It's not necessary because you are using `withAuth` and if you try to use the endpoint without being logged in, the middleware stops you first.

On the other hand, should this endpoint still be enabled once the feature goes into production?

Reply:
I believe it will come in handy if any job fails because the tx costs are too high, or we run out of gas, or we want to give a badge to some users without necessarily automatizing something, like for a `one time` thing.